packages/ir-sdk/fern/apis/ir-types-latest/definition/auth.yml

```diff
@@ -22,13 +23,83 @@ types:
       bearer: BearerAuthScheme
       basic: BasicAuthScheme
       header: HeaderAuthScheme
+      oauth: OAuthScheme
   BearerAuthScheme:
     extends: commons.WithDocs
     properties:
       token: commons.Name
       tokenEnvVar:
         type: optional<EnvironmentVariable>
         docs: The environment variable the SDK should use to read the token.
+  
+  OAuthScheme:
+    extends: commons.WithDocs
+    docs: Note that for the oauth scheme we are currently assuming the resultant token is leveraged as a bearer token, e.g. "Authorization Bearer
+    properties:
+      configuration: OAuthConfiguration
+  OAuthConfiguration:
+    union:
+      authorizationCode: AuthorizationCodeOAuthScheme
+      clientCredentials: ClientCredentialsOAuthScheme
+  OauthAccessTokenFields:
```

Comment:
nit: Maybe best to standardize on either `OAuth*` or `Oauth`. I prefer `OAuth*`, but fine with either way.